cmake/glfw.cmake

```diff
@@ -11,6 +11,9 @@ target_sources(mbgl-glfw
     PRIVATE platform/default/mbgl/util/default_styles.cpp
 )
 
+# Suppress GLFW redefining GLAPIENTRY error.
+SET_SOURCE_FILES_PROPERTIES(platform/glfw/glfw_view.cpp PROPERTIES COMPILE_FLAGS -Wno-error)
```

Comment:
Can we please add some more explanation why we need the hammer (`-Wno-error`) rather than selectively disabling the warning?